### PR TITLE
feat!: drop Node.js 8 support

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes https://github.com/netlify/cli/issues/1568

Not to be merged yet, but opened for visibility

### CODE REVIEW

File: renovate.json5 [hunk 1]
```diff
@@ -9,29 +9,5 @@
...
-        'dot-prop',
-        'execa',
-        'find-up',
-        'globby',
-        'locate-path',
-        'log-symbols',
-        'npm-packlist',
-        'ora',
-        'wrap-ansi',
-        'update-notifier',
```
Comment (ehmicky) [inline renovate.json5:30]: It would be nice to follow up with upgrading those dependencies as well.

**Review by ehmicky — APPROVED** [review]

File: renovate.json5 (hunk 1 above)
Comment (erezrokah) [reply renovate.json5:30]: Yes, I'm counting on renovate to open those PRs so I can examine the changelogs for breaking changes (there might ones not related to the Node.js upgrade).

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "Bret Comnes <bcomnes@gmail.com> (https://bret.io)"
   ],
   "engines": {
-    "node": ">=8.17.0"
+    "node": ">=10"
```
Comment (ehmicky) [inline package.json:14]: Should we specific a specific minor and patch version?

Comment (erezrokah) [reply package.json:14]: From a quick check most of the deps we haven't updated don't require it:
https://github.com/sindresorhus/path-exists/blob/82516fa81d83e75477e517a4ba6c2c8637198521/package.json#L13

Comment (ehmicky) [reply package.json:14]: Unless I'm mistaken, when the minor and patch version are omitted, `engines.node` understands it as "at least the latest 10.*.* available", as opposed to `>=10.0.0` which might be what we mean?

Comment (ehmicky) [reply package.json:14]: I think `engines.node >= 10` would be interpreted by `npm` and `yarn` as `>= 10.0.0`, so it's probably to match what our dependencies are doing :+1:

Comment (erezrokah) [reply package.json:14]: Looks like there is just 1 package that doesn't follow that pattern:
https://github.com/concordancejs/concordance/blob/2bf27e08a2512a586febad3f4eb40548c998fcaa/package.json#L11

We could specify `>=10.18.0`, WDYT?

Comment (ehmicky) [reply package.json:14]: From [the data](https://github.com/netlify/engineering/issues/50#issuecomment-704451333) of Node.js in production (not CLI) from 2 months ago, 93% of users were using `>=10.0.0` and 90% were using `>=10.18.0`. This probably will increase a lot in April when Node 10 is being officially deprecated. 
Based on this, this seems ok to me :+1:

**Pull request merged**